File: 1630-arithmetic-subarrays/1630-arithmetic-subarrays.py
class Solution:
    def checkArithmeticSubarrays(self, nums: List[int], l: List[int], r: List[int]) -> List[bool]:
        def check(arr):
            mn, mx = min(arr), max(arr)
            if (mx-mn) % (len(arr)-1) != 0: return False
            diff = (mx - mn) / (len(arr)-1)
            s = set(arr)
            curr = mn + diff
            while curr < mx:
                if curr not in s: return False
                curr += diff
            return True
        ans = []
        for i in range(len(l)):
            arr = nums[l[i] : r[i]+1]
            ans.append(check(arr))
        return ans

File: 1630-arithmetic-subarrays/1630-arithmetic-subarrays.java
class Solution {
    public List<Boolean> checkArithmeticSubarrays(int[] nums, int[] l, int[] r) {
        List<Boolean> ans = new ArrayList();
        for(int i=0; i<l.length; i++){
            int[] arr = new int[r[i]-l[i]+1];
            for(int j=0; j<arr.length; j++) arr[j] = nums[l[i]+j];
            ans.add(check(arr));
        }
        return ans;
    }
    Boolean check(int[] arr){
        int mn = Integer.MAX_VALUE, mx = Integer.MIN_VALUE;
        Set<Integer> s = new HashSet();
        for(int x : arr){
            mn = Math.min(mn, x);
            mx = Math.max(mx, x);
            s.add(x);
        }
        if((mx-mn) % (arr.length-1) != 0) return false;
        int diff = (mx-mn)/(arr.length-1);
        int curr = mn + diff;
        while(curr < mx){
            if(!s.contains(curr)) return false;
            curr += diff;
        }
        return true;
    }
}